Addresses Media Object mixin issue in Flex mode. #9038.

Martskin opened the pull request:
I refactored the Media Object mixins to try and address #9038.

Martskin committed:
Addresses Media Object mixin issue in Flex mode. #9038.

kball commented:
This makes sense to me, thanks @Martskin!

Would you be able to add a visual test case for this to confirm the behavior & make sure we don't create any regressions? There's information on how to create a visual test here: https://github.com/zurb/foundation-sites/wiki/Testing-Guide#visual-regression-tests

Martskin committed:
Adds Media Object visual test page.

Martskin commented:
@kball 

No problem! I added a visual test page named `media-object-mixins-flex-mode.html`. Lemme know if you want anything else changed or added.

Also, I think this line in the Visual Regression Tests guide may not be necessary anymore?
>If you need to test flexbox mode, change the stylesheet in the <link> tag to foundation-flex.css

ncoden commented on scss/components/_media-object.scss:
```diff
@@ -50,6 +62,27 @@ $mediaobject-image-width-stacked: 100% !default;
     margin-bottom: 0;
   }
+
+  .stack-for-#{$-zf-zero-breakpoint} & {
```
I agree the code would be more cleaner with this in `media-object-selection`. But sadly, `.stack-for-medium .media-object-section` is not the same as `.media-object.stack-for-medium .media-object-section`, bugs can be introduced because `.stack-for-*` is used for other components.

With this selector, if any parent use `stack-for-` for its own purpose, this component will be affected. So i'm moving this to `@mixin media-object-container` to ensure backward compatibility.

(I don't like it neither, it prevent to reduce the specificity 😢  ).

ncoden committed:
fix: prevent side-effect from .stack-for in media-object

Prevent a parent component with `.stack-for-*` to have side effects on `.media-object-section`.

This is a part of a PR and the selector specificity is revert to the one before the PR (0.3.0).
